packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -399,7 +409,26 @@ class _HelperErrorState extends State<_HelperError> with SingleTickerProviderSta
   }
 
   Widget _buildError() {
-    assert(widget.error != null || widget.errorText != null);
+    assert(
+      widget.error != null || widget.errorText != null,
+      'errorText or error must be provided'
+    );
+
+    assert(
+      !(widget.error != null && (widget.errorBuilder != null || widget.errorText != null)),
```

Comment:
we could use 1 less set of parentheses:

```suggestion
      widget.error == null || (widget.errorBuilder == null && widget.errorText == null),
```